src/index.ts

```diff
@@ -0,0 +1,205 @@
+      process.exit();
+    }
+    let account = parachainAccount(id);
+    console.log(
+      `Starting a Collator for parachain ${id}: ${account}, Collator port : ${port} wsPort : ${wsPort}`
+    );
+    startCollator(bin, id, wsPort, port, chain, spec, flags);
+
+    // If it isn't registered yet, register the parachain on the relaychain
+    if (!registeredParachains[id]) {
+      console.log(`Registering Parachain ${id}`);
+
+      // Get the information required to register the parachain on the relay chain.
+      let genesisState;
+      let genesisWasm;
+      try {
+        genesisState = await exportGenesisState(bin, id, chain);
+        genesisWasm = await exportGenesisWasm(bin, chain);
+      } catch (err) {
+        console.error(err);
+        process.exit(1);
+      }
+
+      await registerParachain(relayChainApi, id, genesisWasm, genesisState);
+
+      registeredParachains[id] = true;
+
+      // Allow time for the TX to complete, avoiding nonce issues.
+      // TODO: Handle nonce directly instead of this.
+      if (balance) {
+        await setBalance(relayChainApi, account, balance);
+      }
+    }
+  }
+
+  // Then launch each simple parachain (e.g. an adder-collator)
+  for (const simpleParachain of config.simpleParachains) {
+    const { id, port, balance } = simpleParachain;
+    const bin = resolve(config_dir, simpleParachain.bin);
+    if (!fs.existsSync(bin)) {
+      console.error("Simple parachain binary does not exist: ", bin);
+      process.exit();
+    }
+
+    let account = parachainAccount(id);
+    console.log(`Starting Parachain ${id}: ${account}`);
+    startSimpleCollator(bin, id, spec, port);
+
+    // Get the information required to register the parachain on the relay chain.
+    let genesisState;
+    let genesisWasm;
+    try {
+      // adder-collator does not support `--parachain-id` for export-genesis-state (and it is
+      // not necessary for it anyway), so we don't pass it here.
+      genesisState = await exportGenesisState(bin);
+      genesisWasm = await exportGenesisWasm(bin);
+    } catch (err) {
+      console.error(err);
+      process.exit(1);
+    }
+
+    console.log(`Registering Parachain ${id}`);
+    await registerParachain(relayChainApi, id, genesisWasm, genesisState);
+
+    // Allow time for the TX to complete, avoiding nonce issues.
+    // TODO: Handle nonce directly instead of this.
+    if (balance) {
+      await setBalance(relayChainApi, account, balance);
+    }
+  }
+
+  for (const hrmpChannel of config.hrmpChannels) {
+    await ensureOnboarded(relayChainApi, hrmpChannel.sender);
+    await ensureOnboarded(relayChainApi, hrmpChannel.recipient);
+
+    const { sender, recipient, maxCapacity, maxMessageSize } = hrmpChannel;
+    await establishHrmpChannel(
+      relayChainApi,
+      sender,
+      recipient,
+      maxCapacity,
+      maxMessageSize
+    );
+  }
+}
+
+async function ensureOnboarded(relayChainApi: ApiPromise, paraId:string) {
```

Comment:
Why isn't the paraid an integer?

Reply:
I don't mind changing it.
It's just that in the config, `hrmpChannels.sender` and `hrmpChannels.recipient` are strings

Reply:
Ahh, yeah we should probably change this.

Reply:
updated